server/src/main/java/io/deephaven/server/console/ConsoleServiceGrpcImpl.java

```diff
@@ -80,6 +80,8 @@ public ConsoleServiceGrpcImpl(final TicketRouter ticketRouter,
     public void getConsoleTypes(final GetConsoleTypesRequest request,
             final StreamObserver<GetConsoleTypesResponse> responseObserver) {
         GrpcUtil.rpcWrapper(log, responseObserver, () -> {
+            // for legacy reasons, this method is used before the session is created
+            // TODO: how do we authorize these calls; can we fix the ordering of requests?
             if (!REMOTE_CONSOLE_DISABLED) {
```

Comment:
Hmm. I dropped the todo here because the web ui requests console types before it creates the session. IIRC the console selection from the user was originally required to open the jsapi connection.

Reply:
It is required to create the console session from the JS client, but they can be required to log in first (and soon, in order to use the JS API, will need to authenticate before they can call the method that will fetch this list).